cmd/tnf/addclaim/addclaim.go

```diff
@@ -27,12 +29,205 @@ var (
 		Short: "The test suite generates a \"claim\" file",
 		RunE:  claimUpdate,
 	}
+	claimCompareFiles = &cobra.Command{
+		Use:   "compare",
+		Short: "Compare 2 \"claim\" file",
+		RunE:  claimCompare,
+	}
 )
 
 const (
 	claimFilePermissions = 0o644
 )
 
+type cnistruct []struct {
+	Name    string        "json:\"name\""
+	Plugins []interface{} "json:\"plugins\""
+}
+type Cni struct {
+	Claim struct {
+		Nodes struct {
+			CniPlugins map[string]cnistruct `json:"cniPlugins"`
+		} `json:"nodes"`
+	} `json:"claim"`
+}
+type Csi struct {
+	Claim struct {
+		Nodes struct {
+			CsiDriver interface{} `json:"csiDriver"`
+		} `json:"nodes"`
+	} `json:"claim"`
+}
+
+type HwInfo struct {
+	Claim struct {
+		Nodes struct {
+			NodesHwInfo map[string]interface{} `json:"nodesHwInfo"`
+		} `json:"nodes"`
+	} `json:"claim"`
+}
+
+type RawResult struct {
+	Claim struct {
+		RawResults struct {
+			Cnfcertificationtest struct {
+				Testsuites struct {
+					Testsuite struct {
+						Testcase testcase `json:"testcase"`
+					} `json:"testsuite"`
+				} `json:"testsuites"`
+			} `json:"cnf-certification-test"`
+		} `json:"rawResults"`
+	} `json:"claim"`
+}
+
+//nolint:funlen,gocyclo
```

Comment:
You shouldn't remove the `gocyclo` linter, as it's basically warning that your code needs refactoring due to too many (and probably nested) `if`s and nested loops, which make it harder to review and quite error prone.